unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/ProfanityFiltering/RegexProfanityFilter.cs

```diff
@@ -13,7 +14,7 @@ public RegexProfanityFilter(IProfanityWordProvider wordProvider)
         regex = new Regex(@$"\b({explicitWords})\b|({nonExplicitWords})", RegexOptions.IgnoreCase);
     }
 
-    public string Filter(string message)
+    public async UniTask<string> Filter(string message)
```

Comment:
in case throttling is needed you could segregate the regex into multiple parts (for example totalWords / 5). Then on every frame validate one part of the regex

Reply:
No throttling nor performance is being done on this PR, this is just a refactor to support that change that I'm preparing on another PR

Reply:
https://github.com/decentraland/unity-renderer/pull/2215